internal/envoy/route.go

```diff
@@ -41,21 +42,40 @@ func RouteRoute(r *dag.Route, clusters []*dag.Cluster) *route.Route_Route {
 		)
 	}
 
-	switch len(clusters) {
+	switch len(r.Clusters) {
 	case 1:
 		ra.ClusterSpecifier = &route.RouteAction_Cluster{
-			Cluster: Clustername(clusters[0]),
+			Cluster: Clustername(r.Clusters[0]),
 		}
 	default:
 		ra.ClusterSpecifier = &route.RouteAction_WeightedClusters{
-			WeightedClusters: weightedClusters(clusters),
+			WeightedClusters: weightedClusters(r.Clusters),
 		}
 	}
 	return &route.Route_Route{
 		Route: &ra,
 	}
 }
 
+// hashPolicy returns a slice of hash policies iff at least one of the route's
+// clusters supplied uses the `cookie` load balancing stategy.
+func hashPolicy(r *dag.Route) []*route.RouteAction_HashPolicy {
+	for _, c := range r.Clusters {
+		if c.LoadBalancerStrategy == "cookie" {
+			return []*route.RouteAction_HashPolicy{{
+				PolicySpecifier: &route.RouteAction_HashPolicy_Cookie_{
+					Cookie: &route.RouteAction_HashPolicy_Cookie{
+						Name: "X-Contour-Session-Affinity",
+						Ttl:  duration(0),
```

Comment:
What does this ttl duration do? How long the cookie is valid?

Reply:
If the ttl field is omitted envoy will not emit the cookie - it basically says “use the cookie with this name, I don’t manage it”. We don’t want that, we want envoy to own the cookie so we have to provide a ttl. 

Ttl 0 says to the browser this is a session cookie, it’s lifespan is controlled by the browser, not envoy. It lives as long as the browser wants which seemed like the only reasonable choice.